patches/tModLoader/Terraria/IO/WorldFile.cs.patch

```diff
@@ -188,7 +188,7 @@
  			for (int j = 0; j < Main.npc.Length; j++) {
  				NPC nPC2 = Main.npc[j];
 -				if (nPC2.active && NPCID.Sets.SavesAndLoads[nPC2.type]) {
-+				if (nPC2.active && NPCID.Sets.SavesAndLoads[nPC2.type] && !NPCLoader.IsModNPC(nPC2)) {
++				if (nPC2.active && NPCLoader.SavesAndLoads(nPC2) && !NPCLoader.IsModNPC(nPC2)) {
```

Comment:
I'm wary about changing vanilla saving logic. Is there a reason for a mod to enable saving for a vanilla NPC? Would this break vanilla compat?

Reply:
I suppose you need this to save globals on vanilla NPCs...
I need an example of such in ExampleMod before this can be merged

Reply:
Well this change doesn't really change the behavior of tmodloader as you could already change the previous array.
It would make sense though to only save vanilla NPCs that were intended to be saved, just wanted to follow the logic in place.
If you want a change to it if the array becomes unavailable to modders, feel free to say it
Also for ExampleMod, what kind of example do you want ?

Reply:
I think this just needs reverting. This controls what goes in the vanilla save file, so it should use the vanilla `SavesAndLoads` check, `nPC2.active && NPCID.Sets.SavesAndLoads[nPC2.type] && !NPCLoader.IsModNPC(nPC2)`